src/parser/state_machine/syntax/comment.rs

```diff
@@ -9,15 +9,15 @@ define_state_group!(comment_states_group = {
     comment_start_state <-- ( create_comment; start_token_part; ) {
         b'-' => ( mark_comment_text_end; --> comment_start_dash_state )
         b'>' => ( mark_comment_text_end; emit_current_token?; --> data_state )
-        eof  => ( mark_comment_text_end; emit_current_token_and_eof?; )
+        eof  => ( reconsume in comment_state )
```

Comment:
Can you give an example of a test this fixes? Why is the change necessary? Ditto for the rest of the changes.

Reply:
this is a direct update from the spec: https://html.spec.whatwg.org/multipage/parsing.html#comment-start-state

Reply:
currently playing around with the code to give you concrete testcases

Reply:
FWIW, if the spec says to do it, I'm ok with not having an outright test. I mostly wasn't sure why you'd made the change.

Reply:
yes so I believe if you revert any single change here you should either:

1. no longer line up with the spec
2. break a testcase

I added some clarifying comments because mark_comment_text_end isn't the exact same action as "append current input character"